src/runtime/pay-flow-test.js

```diff
@@ -1054,6 +1054,80 @@ describes.realWin('PayCompleteFlow', (env) => {
     await flow.readyPromise_;
   });
 
+  it('constructs valid flow w/ virtual gift url params even without order id', async () => {
```

Comment:
virtual gifts is TWG, so there's probably no need for a new test.

Reply:
The TypeScript migration is exposing branches that were missing coverage. If code relating to `VIRTUAL_GIFT` purchases is still necessary, I think testing all the branches makes sense. If it's dead code though, we can remove it in a followup PR.